### PR TITLE
Refactor/modular design

### PR DESCRIPTION
**Pull request by sungchun12**
Let's make it easier to evolve building blocks together!

I made a TON of breaking changes to this to do the below.
- Modular file design
- `fst start` now works simply with an optional `--path` flag
- Rerun with `dbt test` if first `dbt build` command did not run tests
- Kept in the platform agnostic polling mechanism 
- No need to use nested git clones once released to pypi

image

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -55,38 +56,64 @@ python3 -m pip install --upgrade pip setuptools wheel
 source venv/bin/activate
 pip3 install -r requirements.txt
 pip3 install -e ../ # installing the fst package locally
-# fst start --file-path <file path>
+dbt build # Optional, so that upstream dependencies are created
 code .
-fst start --file-path models/customers.sql
+fst start
```
Comment (xtomflo) [inline README.md:61]: awesome!

**Review by xtomflo — APPROVED** [review]
Looks like an awesome piece of work to me! Tested it and the results look as expected. 

To note:
The dbt build time is big, at ~20 seconds. Although this is likely my local issue as it persists when going back to earlier commits as well. 

At some point I've had new yaml file be generated for models which were already described in schema.yml, but I couldn't reproduce it. It must be some edge case

File: fst/main.py [hunk 2]
```diff
@@ -1,84 +1,34 @@
...
+    help="dbt project root directory. Defaults to current working directory.",
 )
-def restart(file_path):
-    if fst_query.observer:
-        fst_query.observer.stop()
-        fst_query.observer.join()
-        click.echo("Stopped watching the directory.")
-    else:
-        click.echo("No observer was running. Starting a new one.")
+def start(path):
```
[the remark this answers is not in the document]
Comment (sungchun12) [reply fst/main.py:26]: It takes it in as the default, it doesn't return anything as it simply passes it to the `project_dir` variable

**Comment by sungchun12** [comment]
> To note: The dbt build time is big, at ~20 seconds. Although this is likely my local issue as it persists when going back to earlier commits as well.
> 
> At some point I've had new yaml file be generated for models which were already described in schema.yml, but I couldn't reproduce it. It must be some edge case

Yep, I've had this happen rarely, but I can see that. For the `dbt build`, that feels like a WSL issue because compilation and runtime shouldn't be that long.

**Pull request merged**